### PR TITLE
Updated plugins and Jenkins Core - removed analysis-core

### PR DESCRIPTION
**Pull request by alxsap**
This PR is blocked, because some of our pipelines still require the `analysis-collector` plugin.
`analysis-collector` has been removed from the Jenkins Update site as it has been deprecated for a longer time. We should not try to "get it working" with the old plugin dependencies and update other plugins. I'm also not sure if a new Jenkins version (without plugin updates) would work.

### CODE REVIEW

**Comment by alxsap** [comment]
Switched to non-LTS Jenkins since we take the latest plugins also from non-LTS which caused issues.
If we decide to stay with LTS we need to use the LTS update site as well for the plugins.

File: jenkinsfile-runner-base-image/packager-config.yml [hunk 1]
```diff
@@ -21,15 +21,15 @@ buildSettings:
...
+      base: "jenkins/jenkins:2.291-alpine"
       tag: "jenkinsfile-runner-base-local"
       build: true
 
 war:
   groupId: "org.jenkins-ci.main"
   artifactId: "jenkins-war"
   source:
-    version: "2.277.4"
+    version: "2.291"
```
Comment (alxsap) [inline jenkinsfile-runner-base-image/packager-config.yml:32]: With this Jenkins version the build fails with:
```
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for org.jenkins-ci.modules:ssh-cli-auth:jar is missing. @ line 51, column 17
[ERROR] 'dependencies.dependency.version' for org.jenkins-ci.modules:sshd:jar is missing. @ line 75, column 17
```

Comment (alxsap) [reply jenkinsfile-runner-base-image/packager-config.yml]: switched to LTS again, including plugins.

**Comment by alxsap** [comment]
> Switched to non-LTS Jenkins since we take the latest plugins also from non-LTS which caused issues.
> If we decide to stay with LTS we need to use the LTS update site as well for the plugins.

Switched back to LTS, including plugins.

**Comment by alxsap** [comment]
Reference pipelines are successful now.

**Comment by srinikitha09** [comment]
I think, we might not even need `badge` plugin since we are not assigning any badges. What do you think @radsoulbeard ?

**Comment by radsoulbeard** [comment]
> I think, we might not even need `badge` plugin since we are not assigning any badges. What do you think @radsoulbeard ?
> 
> Currently, we are not providing any email notification to the users regarding the status of the failure or success of a build. Hence, in that case, we can also remove `email-ext` plugin.
> 
> @alxsap Is there a possibility that you can test our reference pipelines without these plugins?

I found an addBadge statement in the Jenkins-library. It is in the step checkForLegacyConfiguration. This step is is called in piperPipelineStageInit and we use this init Stage in GPP for CAP.

**Comment by srinikitha09** [comment]
> > > I think, we might not even need `badge` plugin since we are not assigning any badges. What do you think @radsoulbeard ?
> > > Currently, we are not providing any email notification to the users regarding the status of the failure or success of a build. Hence, in that case, we can also remove `email-ext` plugin.
> > > @alxsap Is there a possibility that you can test our reference pipelines without these plugins?
> > 
> > 
> > I found an addBadge statement in the Jenkins-library. It is in the step checkForLegacyConfiguration. This step is is called in piperPipelineStageInit and we use this init Stage in GPP for CAP.
> 
> I have just ran a gpp pipeline with GettingStartedBookShop project [here](https://cicddemo.cicd.cfapps.sap.hana.ondemand.com/service/v2/jobs/gpp-cap/builds/9/log?stageId=11) wherein, I do not see that step being called.

Ok, we might need this badge plugin, incase, if there are changes in configuration for stage/step which are not supported anymore.

**Comment by alxsap** [comment]
@srinikitha09 @valiparsa @radsoulbeard 
I'll remove the two plugins to get a new image for testing it... We can add them afterwards, if they are required.

**Comment by alxsap** [comment]
> > @alxsap Is there a possibility that you can test our reference pipelines without these plugins?
> 
> There is a possiblity to use special Jenkinsfilerunner docker image for pipeline validation. You can trigger `clientPreparationPipeline` and use any Jenkinsdockerimage for this field `jenkinsfileRunnerImage`. If you leave this field empty the default one will be taken into account.

Testing the image from f21aa19 via validation pipeline run 241.

**Comment by alxsap** [comment]
Validation was successful with JFR image version `TEST_210512_f21aa19`.
@radsoulbeard @srinikitha09 Shall we keep email-ext and badge plugin out? At least the 3 validation pipelines worked. Not sure if all customer combinations/configurations will work.

**Comment by alxsap** [comment]
Confirmed

**Pull request merged**